woocommerce/woocommerce-android
# Issue/9612 discoun edit add new layout

Closes: #9612 


### Description


This PR is part of M7 of the Orders with Coupons project. 

The designs for reference can be found pecCkj-RY-p2

Please note that a few changes were made, after some discussion. We decided to skip the Order Total line for now as well as update some of the behaviour/ calculations were made to match iOS. Context here: p1692102894526579-slack-C025A8VV728

Please note that we also skipped M6 (the collapsing Product card for the moment.

### Testing instructions


1. Go to Orders
2. create a new order
3. add a product
4. click on the product
5. click on Add Discount
6. Add a discount
7. switch between % and $
8. make sure the Calculated percentage and amount are correctly displayed
9. click on Done
10. make sure the discount you added is displayed in the Discounts Total line
11. increase and decrease the product count and make sure the discount amount does NOT changes
12. add other products and discount to the new products. 
13. make sure the discount total line sums up all the discounts added

### Images/gif


![Screenshot 2023-08-23 at 5 12 49 PM](https://github.com/woocommerce/woocommerce-android/assets/30724184/6879f5f6-a191-41c5-bfd9-e4056b7bf4f1)
